core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestRemoveDistinctAggregationForGroupedAggregation.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import io.trino.testing.StandaloneQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.parallel.Execution;
+
+import static io.airlift.testing.Closeables.closeAllRuntimeException;
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+public class TestRemoveDistinctAggregationForGroupedAggregation
+{
+    protected StandaloneQueryRunner queryRunner;
+
+    @BeforeAll
+    public final void initRemoveDistinctAggregationForGroupedAggregation()
+    {
+        queryRunner = new StandaloneQueryRunner(TEST_SESSION);
+    }
+
+    @AfterAll
+    public final void destroyRemoveDistinctAggregationForGroupedAggregation()
+    {
+        closeAllRuntimeException(queryRunner);
+        queryRunner = null;
+    }
+
+    private static final String QUERY = "" +
+            "SELECT DISTINCT orderstatus, COUNT(orderkey)\n" +
+            "FROM (\n" +
+            "  VALUES\n" +
+            "    ( 1, 'Done', '1996-01-02'),\n" +
+            "    ( 2, 'Pending', '1996-12-01'),\n" +
+            "    ( 3, 'Done', '1993-10-14'),\n" +
+            "    ( 4, 'Pending', '1995-10-11'),\n" +
+            "    ( 5, 'Done', '1994-07-30'),\n" +
+            "    ( 6, 'Done', '1992-02-21'),\n" +
+            "    ( 7, 'Pending', '1996-01-10'),\n" +
+            "    (32, 'Pending', '1995-07-16'),\n" +
+            "    (33, 'Done', '1993-10-27'),\n" +
+            "    (34, 'Done', '1998-07-21')\n" +
+            ") AS orders (orderkey, orderstatus, orderdate)\n" +
+            "GROUP BY orderstatus";
+
+    @Test
+    public void testRemoveDistinctAggregationForGroupedAggregation()
```

Comment:
Let's assert over the plan shape using the plan assertions instead of testing the SQL plan string. Examples can be found in other tests like `TestRemoveAggregationInSemiJoin` or `TestRemoveEmptyGlobalAggregation`.

Let's also add tests that assert the plan shape over:
- distinct aggregations where one of the source aggregation grouping keys is not present in the distinct (should not fire)
- project and filter nodes are present between the source and distinct aggregation